File: services/event_service/app/database/models/status_track.py
from sqlalchemy import Column, ForeignKey, Integer

from database.schemas.status import StatusTrackSchema

from . import base


class StatusTrack(base.BaseModel):
    __tablename__ = "status_track"

    event_id = Column(
        Integer, ForeignKey("event.id", ondelete="CASCADE"), nullable=False
    )
    status_id = Column(
        Integer, ForeignKey("status.id", ondelete="CASCADE"), nullable=False
    )

    def to_read_model(self) -> StatusTrackSchema:
        return StatusTrackSchema(
            id=self.id,
            event_id=self.event_id,
            status_id=self.status_id,
        )

    @classmethod
    def convert_scheme(cls):
        return StatusTrackSchema

File: services/event_service_golang/internal/models/status_track.go
package models

type StatusTrack struct {
	tableName struct{} `pg:"status_track"`
	TrackID   int      `pg:"track_id,pk"`
	Track     *Track   `pg:"rel:has-one"`

	StatusID int     `pg:"status_id,pk"`
	Status   *Status `pg:"rel:has-one"`
}
